[18.0][MIG] l10n_jp_summary_invoice: Migration to 18.0

AungKoKoLin1997 opened the pull request:
Depends on https://github.com/OCA/account-invoicing/pull/2041

@qrtl QT5088

AungKoKoLin1997 committed:
[ADD] l10n_jp_summary_invoice

Commit:
[UPD] Update l10n_jp_summary_invoice.pot

OCA-git-bot committed:
[BOT] post-merge updates

kanda999 committed:
[16.0][IMP] report_alternative_layout: Adding useful options

- Moved the option for selecting the alternative layout to the paper format settings
- Added an option to include the partner’s address in the header when using the alternative layout
- Added an option to display the document name on the report
- Added an option to display the report date on the report
- Added an option to display the report date label on the report
- Refactored some code
- Updated translations

OCA-git-bot committed:
[BOT] post-merge updates

AungKoKoLin1997 committed:
[IMP] l10n_jp_summary_invoice: pre-commit auto fixes

AungKoKoLin1997 committed:
[MIG] l10n_jp_summary_invoice: Migration to 18.0

AungKoKoLin1997 committed:
[FIX] l10n_jp_summary_invoice: fix wrong tax_total calculation

yostashiro commented:
/ocabot migration l10n_jp_summary_invoice

AungKoKoLin1997 committed:
[FIX] l10n_jp_summary_invoice: Correct handling of credit notes

Previously, credit notes were incorrectly treated as positive amounts
in reporting and tax computation, just like regular invoices. This
commit addresses the issue by introducing the `signed_quantity` field
on `account.move.line` and refactoring the `_compute_tax_totals()`
method to delegate tax computation to Odoo’s standard logic in
`account.move`.

Minor improvements were also made to the generated tax adjustment
entries, including setting the correct due date and enhancing the
reference field.

yostashiro commented:
@AungKoKoLin1997 account_billing migration PR has been stale. Shall we create another one ourselves?

AungKoKoLin1997 commented:
> account_billing migration PR has been stale. Shall we create another one ourselves?

@yostashiro I will create new one if it is still stale on next Monday.

AungKoKoLin1997 committed:
[IMP] l10n_jp_summary_invoice: improve _get_move()

AungKoKoLin1997 committed:
[IMP] l10n_jp_summary_invoice: remove unncessary code

AungKoKoLin1997 committed:
[FIX] l10n_jp_summary_invoice: adjust report layout

Co-authored-by: Yoshi Tashiro <tashiro@quartile.co>

kanda999 reviewed (approved):
Functional review: it works as exptected.

yostashiro commented:
/ocabot rebase

AungKoKoLin1997 commented:
@kanda999 I add a new commit for the report.
Before the fix if we use together with [account_billing_portal](https://github.com/OCA/account-invoicing/pull/2186), the UI is broken.
<img width="1331" height="828" alt="Screenshot 2025-12-23 165142" src="https://github.com/user-attachments/assets/1bc8d4f9-ba1d-4be5-bfe6-258cd0b113db" />

After the fix, the UI is displayed correctly.

<img width="1334" height="894" alt="Screenshot 2025-12-23 165126" src="https://github.com/user-attachments/assets/3aaa5b23-7d20-4dca-83f4-cf9aa6074770" />

yostashiro reviewed (commented):
@AungKoKoLin1997 I've removed the sale dependency and fixed a layout issue (with some other minor adjustments) in my last commit.

Your last update was hiding the upper part of the untaxed amount line, so I've removed the negative margin (`mt-n3`), and then removed the top border of the tax table with `border-top-0` for a cleaner look.
<img width="802" height="155" alt="image" src="https://github.com/user-attachments/assets/56e54823-58f1-4715-a1bb-92247c98ba90" />

Can you please reorganize the commits?

yostashiro committed:
[IMP] l10n_jp_summary_invoice: add billing amount fields and improve UI

- Add amount_untaxed, amount_tax, amount_total computed fields
- Change tax_totals from Binary to Json field (stored)
- Display tax totals in form view using account-tax-totals-field widget
- Add amount fields to list views with sum aggregation
- Add "Print Summary Invoice" button to form header
- Add currency_id filter and group-by option in search view
- Style report table header with light gray background

yostashiro commented:
I've added another commit to add several improvements.

For visually noticeable changes...
<img width="719" height="200" alt="image" src="https://github.com/user-attachments/assets/3a19698b-ff8c-4903-99c9-c827ed234024" />
<img width="973" height="292" alt="image" src="https://github.com/user-attachments/assets/5b07d256-71cf-4e7e-b161-3391c2bd58fa" />
<img width="523" height="341" alt="image" src="https://github.com/user-attachments/assets/2ab99c80-a3b3-4ec1-8317-26320d8ca336" />
<img width="411" height="207" alt="image" src="https://github.com/user-attachments/assets/483dac6f-a651-43ce-aa03-4777f9c62de5" />
<img width="750" height="153" alt="image" src="https://github.com/user-attachments/assets/230843a2-f477-4ec1-8879-87b45a92e1be" />

kanda999 reviewed (approved):
Functional and code review.

AungKoKoLin1997 commented:
In my new commit, I assign the tax_totals with the format that suitable with JS widget instead of assigning `False` or `None` or `{}`.
If not, this JS code https://github.com/odoo/odoo/blob/e60b443136f8713fea8af8e69df04e0b5a858cad/addons/account/static/src/components/tax_totals/tax_totals.js#L162-L164 will remain the default value that is assigned in `setup()`.
https://github.com/odoo/odoo/blob/e60b443136f8713fea8af8e69df04e0b5a858cad/addons/account/static/src/components/tax_totals/tax_totals.js#L131

Then, the template will rendered and cause the issue.
https://github.com/odoo/odoo/blob/e60b443136f8713fea8af8e69df04e0b5a858cad/addons/account/static/src/components/tax_totals/tax_totals.xml#L42-L44

May be it is better we can improve the odoo standard JS behavior but adjusting the behavior in current PR is faster to proceed since odoo review team has slow response and I am not expert at JS.

yostashiro commented:
@AungKoKoLin1997 I think your last fix is a valid approach. Please go ahead and clean up the commit history. 🙏

AungKoKoLin1997 commented:
> @AungKoKoLin1997 I think your last fix is a valid approach. Please go ahead and clean up the commit history. 🙏

@yostashiro Done!

yostashiro reviewed (approved):
Code review. LGTM. 👍

yostashiro commented:
/ocabot merge nobump
